src/main/java/weijie/duke/presenters/TasksPresenter.java

```diff
@@ -0,0 +1,50 @@
+package weijie.duke.presenters;
+
+import weijie.duke.commands.ITaskCommand;
+import weijie.duke.commands.TaskCommandFactory;
+import weijie.duke.exceptions.DukeException;
+import weijie.duke.responses.TaskResponse;
+import weijie.duke.views.Ui;
+
+
+public class TasksPresenter {
+    private Ui ui;
+    private TaskCommandFactory factory;
+
+    public TasksPresenter(Ui ui, TaskCommandFactory factory) {
+        this.ui = ui;
+        this.factory = factory;
+    }
+
+    /**
+     * <p>
+     *     Starts the main flow of the app. Presenter will start to listen for user input from the Ui, and will
+     *     create appropriate Commands and invoke them, returning any TaskResponses to the Ui to be displayed.
+     * </p>
+     */
+    public void run() {
+        ui.startDisplay();
+
+        while (true) {
+            String input = ui.readCommand();
+
+            if (input.equals("bye")) {
```

Comment:
Will be better to add a ByeCommand or something similar such that the loop takes in a boolean variable and this variable should always be true unless the ByeCommand is entered.